File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/common/schedulers/ImmediateSchedulerProvider.java
package com.benoitquenaudon.tvfoot.red.app.common.schedulers;

import com.benoitquenaudon.tvfoot.red.util.MethodsAreNonnullByDefault;
import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

/**
 * Implementation of the {@link BaseSchedulerProvider} making all {@link Scheduler}s immediate.
 */
@MethodsAreNonnullByDefault public class ImmediateSchedulerProvider
    implements BaseSchedulerProvider {

  @Override public Scheduler computation() {
    return Schedulers.trampoline();
  }

  @Override public Scheduler io() {
    return Schedulers.trampoline();
  }

  @Override public Scheduler ui() {
    return Schedulers.trampoline();
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/common/schedulers/ImmediateSchedulerProvider.kt
package com.benoitquenaudon.tvfoot.red.app.common.schedulers

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers
import javax.inject.Singleton

/**
 * Implementation of the [BaseSchedulerProvider] making all [Scheduler]s immediate.
 */
@Singleton
class ImmediateSchedulerProvider : BaseSchedulerProvider {
  override fun computation(): Scheduler {
    return Schedulers.trampoline()
  }

  override fun io(): Scheduler {
    return Schedulers.trampoline()
  }

  override fun ui(): Scheduler {
    return Schedulers.trampoline()
  }
}
